kod/object/item.kod

```diff
@@ -633,6 +690,46 @@ messages:
       return FALSE;
    }
 
+   GetMagicRarityGradeCount()
+   "Checks plItem_attributes for HasRarity() = TRUE."
+   "For each attribute with HasRarity = TRUE,  add 1 to count"
+   "Return either viRarity or count, whichever is larger"
+   "This ensures the higher rarity is chosen if in the case "
+   "of competing rarities for an item."
+   {
+
+      local i, iNum, oItemAtt, count;
+      count = 0;
+
+      if Send(self,@HasAnyAttribute)
+      {
+         for i in plItem_attributes
+         {
+            iNum = Send(self,@GetNumFromCompound,#compound=First(i));
+            oItemAtt = Send(SYS,@FindItemAttByNum,#num=iNum);
+
+            if Send(oItemAtt,@HasRarity)
```

Comment:
"HasRarity" is kind of an odd construction.  I can think of two possibilities:

1. IsRare() or AddsToRarity() would be clearer if this is going to be a boolean.
2. You could have this function return an int, and sum up all the rarity values.  That would allow an attribute to contribute more than 1 if that's ever wanted.  But if that's overkill, a boolean is also fine.

Reply:
Refactored `HasRarity` as suggested in option 2 to `GetRarityCountModifier`

It now returns a int of local variable count.  By default it returns count=0 unless there are modifiers in the future for an item attribute.

I also changed the description to say it is a modifier and could be either positive or negative.  Perhaps at some point in the future we will want to add an inferior item category.

Reply:
Cool!